Licensing fixes

illwieckz opened the pull request:
See Unvanquished/Unvanquished#1030 for a similar PR on Unvanquished side.

This PR is created to submit licensing fixes and rewording to Daemon source code.

The first commit is about brand new files by @gimhael introduced in commit 0bb3550ce16ab7b165a88ebb365edf6b2837d1f5 that has Tr3b  GPL header just because the skeleton was copypasted (header and main block) but everything is original work.

See that talk:

```
2017-09-06 09:37:16 +0200 <illwieckz>	I have a question though, why the depthtile*glsl / lighttile*glsl files have Tr3b/XreaL/GPL copyright header if they are brand new? Can't we put another copyright and license on them?
2017-09-06 09:37:38 +0200 <illwieckz>	they were introduced in 0bb3550ce16ab7b165a88ebb365edf6b2837d1f5 (Daemon tree)
2017-09-06 09:40:27 +0200 <illwieckz>	https://github.com/DaemonEngine/Daemon/commit/0bb3550ce16ab7b165a88ebb365edf6b2837d1f5
2017-09-06 09:40:57 +0200 <illwieckz>	gimhael_, what are your thoughts about this? ↑
2017-09-06 09:42:33 +0200 <gimhael_>	iirc i copied some other simple shader files and modified them, that's why the original copyright is there
2017-09-06 09:44:14 +0200 <illwieckz>	okie
2017-09-06 09:45:17 +0200 <illwieckz>	but was there any trace of the previous file (except header) when comitted for the first time?
2017-09-06 09:46:21 +0200 <illwieckz>	if the file was entirely rewritten, you can put the copyright you want
2017-09-06 09:47:18 +0200 <gimhael_>	i think just the structural parts, like "void main() {", the shader logic was all new
2017-09-06 09:50:04 +0200 <illwieckz>	so the only original thing is the header itself…
2017-09-06 09:52:15 +0200 <gimhael_>	yes, i think we could replace it
2017-09-06 09:53:52 +0200 <illwieckz>	I've seen brand new file using BSD © Daemon Developpers like this: https://github.com/DaemonEngine/Daemon/blob/master/src/engine/framework/Application.cpp
```

So @gimhael obviously have the right to _not attribute copyright to Tr3b_ on these files, and is able to chose the license he wants and the copyright attribution he wants. By the way, since this PR changed the license to BSD 3 clause and copyright attribution to Daemon Developers, we need an explicit acknowledgement by gimhael to merge this. So gimhael if you prefer another license or copyright attribution it's very ok, tell it in comments, but whatever you prefer it's probably better for you to not attribute copyright to someone not involved. :wink:

So:

> gimhael do you give your permission to relicense to BSD 3 Clause and attribute to Daemon Developers the files I discover to be brand new original work from you? Or can you define your licensing and attribution preference?

This PR (or subsequent ones) is likely to received header rewording to unify them and make grepping easier (as Unvanquished/Unvanquished#1030 does).

On a similar topic of gimhael's shaders, it looks like some brand new original code by @cmf028 is copypasting the GPL header with Zenimax clause. I will investigate it more, but if it appears these files are brand new original work by Fuma, with his permission we can set another license (like BSD 3 Clause), another attribution (Daemon Developers) or whatever he prefers, and remove the Zenimax clause on them if they were not involved at all in that work. I'll dig into it more further.

Note that as stated in [COPYING.txt](https://github.com/DaemonEngine/Daemon/blob/master/COPYING.txt) file, the default license for Daemon is the BSD 3 Clause, and files under GPL are referenced as exception, that why I suggest BSD 3 Clause by default unless otherwise stated when a relicense is possible.

gimhael commented:
Yes, I agree to the proposed license change. Please go ahead and merge the changes.

illwieckz commented:
@kangz see the commit named “BSD fxaa shader licensing (files by @kangz)”, is that true theses files are brand new from you ? If yes do you agree to switch the license to BSD 3 Clause ?

Note that we also need permission by @cmf028 to relicense @gimhael and @kangz shaders.

Kangz commented:
LGTM for the fxaa shaders on my side.

illwieckz commented:
@cmf028 any comment?

Note that I would ask you the same for these files:

- [src/engine/botlib/bot_api.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_api.h)
- [src/engine/botlib/bot_convert.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_convert.cpp)
- [src/engine/botlib/bot_convert.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_convert.h)
- [src/engine/botlib/bot_debug.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_debug.cpp)
- [src/engine/botlib/bot_debug.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_debug.h)
- [src/engine/botlib/bot_load.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_load.cpp)
- [src/engine/botlib/bot_local.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_local.cpp)
- [src/engine/botlib/bot_local.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_local.h)
- [src/engine/botlib/bot_nav.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_nav.h)
- [src/engine/botlib/bot_navdraw.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_navdraw.h)
- [src/engine/botlib/bot_nav_edit.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_nav_edit.cpp)
- [src/engine/botlib/bot_types.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_types.cpp)

It looks like these bot-related files are brand new from you, if you're OK and if contributors are OK we would be able to remove the useless Zenimax Clause and also put a BSD license on them.

illwieckz commented:
@DolceTriade this file looks to be brand new file from you:

- [src/engine/null/null_renderer.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/null/null_renderer.cpp) (see [`f854e45`](https://github.com/DaemonEngine/Daemon/commit/f854e454ff5ee771c491b1e5e5c86bd0d518c400#diff-d2a454ec3022c55a7061696c2ffd8c32))

Do you confirm?

cmf028 commented:
> 
> 
> @cmf028 any comment?
> 
> Note that I would ask you the same for these files:
> 
>     * [src/engine/botlib/bot_api.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_api.h)
> 
>     * [src/engine/botlib/bot_convert.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_convert.cpp)
> 
>     * [src/engine/botlib/bot_convert.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_convert.h)
> 
>     * [src/engine/botlib/bot_debug.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_debug.cpp)
> 
>     * [src/engine/botlib/bot_debug.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_debug.h)
> 
>     * [src/engine/botlib/bot_load.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_load.cpp)
> 
>     * [src/engine/botlib/bot_local.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_local.cpp)
> 
>     * [src/engine/botlib/bot_local.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_local.h)
> 
>     * [src/engine/botlib/bot_nav.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_nav.h)
> 
>     * [src/engine/botlib/bot_navdraw.h](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_navdraw.h)
> 
>     * [src/engine/botlib/bot_nav_edit.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_nav_edit.cpp)
> 
>     * [src/engine/botlib/bot_types.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/botlib/bot_types.cpp)
> 
> 
> It looks like these bot-related files are brand new from you, if you're OK and if contributors are OK we would be able to remove the useless Zenimax Clause and also put a BSD license on them.

I wrote all of these myself. The Zenimax clause was an artifact of copy/paste the license from other source files. I'm fine with putting a BSD license on them.

illwieckz commented:
great :+1:

illwieckz commented:
After a deep review of all `botlib` commits that are not commits by Fuma neither merge commits, I confirm they all are trivial doing trivial things like `Replace NULL and 0-pointer with nullptr.` or `Replace MIN and MAX with std::min and std::max where possible.`.

There was one commit that was doing things more than that but this commit is now reverted since being buggy in any way.

So we can relicense `botlib` files.

List of trivial commits compatible with a relicense of `botlib` stuff:

- 07f33746300ba9d1554c6f21473d5b1db92afa6c
```
Author: dimhotepus
Date:   Mon Mar 7 16:45:22 2016 +0300

    Apply GNU compilers specific pragmas under GNU
```

- 0f8fee8aa12f601aaa8195c8ec037d726126c77b
```
Author: TimePath
Date:   Fri Apr 3 17:04:54 2015 +1100

    Suppress many instances of writable-strings
```

- 10ef9ec1d155cc8a2b99e6caa61e140826f0512c
```
Author: Tim
Date:   Thu Nov 15 00:54:43 2018 +0100

    Use `default` for trivial default constructors.
```

- 13041a86b0c33e165ce1750db45c34dd84028b77
```
Author: Darren Salt
Date:   Wed Dec 4 17:52:54 2013 +0000

    Replace MIN and MAX with std::min and std::max where possible.
```

- 190fd6d1ab62b481b79832a094f663e89c94db62
```
Author: Corentin Wallez
Date:   Mon Apr 13 02:53:47 2015 +0200

    Put vector in the Math namespace, add VecX to the prelude
```

- 1eb96ce1038474d667f402a46b4c1c8267c1b0af
```
Author: Tim
Date:   Wed Nov 14 02:51:24 2018 +0100

    Replace NULL and 0-pointer with nullptr.
```

- 29384165ad47e075bd3231091c2646f3d29125d7
```
Author: TimePath
Date:   Sat Jan 16 23:02:23 2016 +1100

    s/#define/static const/
```

- 2fd8e807db6ea2346bf087bea70e642884af3890
```
Author: Darren Salt
Date:   Tue Feb 5 03:59:22 2013 +0000

    Nav mesh debug draw compile fixes.
```

- 347ac550561e3580db032819407985f4bd9539fc
```
Author: slipher
Date:   Thu Dec 27 16:41:16 2018 -0600

    Fix some spelling errors in comments
```

- 34c68dfcc3c8bccee235dfbdd91ae4f187b0aac3
```
Author: slipher
Date:   Fri Jul 13 06:42:02 2018 -0700

    Replace Format("%s, s) with Format(s) in all Str::Format-based APIs.
```

- 3641054ddc533da9b4a001da65d520d75d4ac3ff
```
Author: TimePath
Date:   Sat Jan 16 16:21:26 2016 +1100

    Classier enums
```

- 38eaa7006629b00eb46858adae2bdf5eb9691777
```
Author: Mattia Basaglia
Date:   Thu Aug 6 14:00:45 2015 +0200

    Remove color macros
```

- 398e83f2ef9208859b22220f31a196dbaf5bb142
```
Author: TimePath
Date:   Thu Jan 21 10:51:56 2016 +1100

    Remove redundant log constant prefixes
```

- 3fe3afc8721fabe62a760a13274c06600e8b2cbf
```
Author: TimePath
Date:   Thu Jan 21 16:18:16 2016 +1100

    Logging part 3
```

- 404fc4e637f460504ccd8c9059481ce2bbf5b230
```
Author: TimePath
Date:   Thu Jan 21 13:21:15 2016 +1100

    Logging part 1
```

- 4d3fc6005a1984ccf8e0900dbb030d7812073a3b
```
Author: Corentin Wallez
Date:   Wed Jan 29 00:32:18 2014 +0100

    Remove all the CVAR_ARCHIVE flags
```

- 57fc335909c09bff27ead180c81844f3c2281919
```
Author: Corentin Wallez
Date:   Mon Feb 24 21:58:30 2014 +0100

    Translate bot syscalls
```

- 603b688dad0655bb91cf999655cc8bc6d79e0196
```
Author: TimePath
Date:   Wed Apr 15 12:34:00 2015 +1000

    Replace NULL with nullptr in code
```

- 631bb5c0b7eb4985707cbc3a87ebc96fd3124f63
```
Author: Darren Salt
Date:   Wed Jan 30 13:55:11 2013 +0000

    POLYFLAGS_SWIM was missing.
```

- 6a0604ea8a619daf8bd81be62f0eb92601cbcd79
```
Author: Amanieu d'Antras
Date:   Sun Jul 6 18:46:41 2014 +0200

    Various cleanups
```

- 726b17a7c3153f803183b3f8faa36522f95abae7
```
Author: Darren Salt
Date:   Sun Jun 30 01:23:47 2013 +0100

    Make navmesh connection editing a little easier.
```

- 730e9a59779ed472cb3433736ce41cb58e1bc755
```
Author: TimePath
Date:   Wed Apr 15 14:32:36 2015 +1000

    Replace func(void) with func()
```

- 7908ce08607590b8cf74561327fd67c59a54eecd
```
Author: Viech
Date:   Wed Jun 4 22:13:06 2014 +0200

    Add a skip mask to all traces, which allows to selectively ignore content types.
```

- 84057b979094edb4100338d76f9a79751e13b41b
```
Author: slipher
Date:   Thu Dec 27 16:45:31 2018 -0600

    Fix some spelling errors and function names in log messages
```

- 8d7c39da54438c7145acb3a144e211ffc88ce3ea
```
Author: slipher
Date:   Wed Dec 26 19:53:07 2018 -0600

    Delete unused variable
```

- 8e00403189ff32d1c1acb216e94796b8749da4c7
```
Author: Thomas Debesse
Date:   Tue May 15 21:08:39 2018 +0200

    fix minor warning misspell
```

- 8f16cf3b66b00bd7ef8a723e8a06c2e12c4a6695
```
Author: Viech
Date:   Mon Oct 28 23:29:59 2013 +0100

    Move bg_* files from "game" to "shared".
```

- 90ad4c1e386778058cb0bf658d99a8edd3bf8f33
```
Author: Corentin Wallez
Date:   Fri Mar 27 03:47:28 2015 +0100

    Add include dirs, make all includes use absolute paths
```

- a05a9bc4334b0cf25e323ebf2cd4f2f8ed92d36c
```
Author: TimePath
Date:   Wed Apr 15 12:05:24 2015 +1000

    Remove the qboolean enum
```

- ae5bf4822311dee0fd1cb23547c69971d622f102
```
Author: Amanieu d'Antras
Date:   Sat Dec 14 23:36:33 2013 +0000

    Consolidate headers:
```

- b17b08ad05dd8660e4e7bc912bdcf6e75e77aa6c
```
Author: TimePath
Date:   Fri Jan 15 20:30:46 2016 +1100

    using typedef = using
```

- bac032c50aef7b2c0fbb7d9218acd269dd7ca0c9
```
Author: Mattia Basaglia
Date:   Fri Aug 14 10:38:42 2015 +0200

    Remove Color::NamedString
```

- bc54ac2916f7f245379ad32f5b9b2b94f620ba9d
```
Author: maek
Date:   Sun Oct 7 11:35:59 2018 +0100

    Other minor fixes
```

- bec64d007d51b54e158db691b46a69facdaf4576
```
Author: Amanieu d'Antras
Date:   Wed Apr 9 23:39:24 2014 +0100

    Change all #ifdefs to use the new BUILD_* target macros
```

- cb6865cfd79ddef49550b05cebbcf341cd1c5fb2
```
Author: Darren Salt
Date:   Wed Dec 11 14:09:45 2013 +0000

    Replace min+max pairs with Maths::clamp.
```

- d09abd5cb6657d2f42b296ec66febc88ee011e4d
```
Author: Corentin Wallez
Date:   Fri Aug 7 06:03:29 2015 +0200

    Fix all regular warnings in the engine except in renderer
```

- de443dc185656d2ca581c848f3ee189351a05833
```
Author: TimePath
Date:   Wed Jul 15 22:03:42 2015 +1000

    Make PNaCl a first class CMake toolchain; split engine and game build scripts
```

- eec8091fbb846dc4a265c21d97ec171a485beca3
```
Author: Tim
Date:   Sat Apr 6 04:27:07 2013 +0200

    include bg_public.h in bot_debug.h, so we can remove it from the rest of the client
```

- f0768e464c1ff369ce77e4853278a0b760464290
```
Author: Darren Salt
Date:   Sun Jun 30 01:23:29 2013 +0100

    Fix a linkage warning.
```

There is no commits incompatible with a relicense of `botlib` stuff.

cmf028 commented:
> 
> 
> @Kangz have a look at the commit named “_BSD fxaa shader licensing (files by @Kangz)_”, is that true theses files are brand new from you ? If yes do you agree to switch the license to BSD 3 Clause ?
> 
> We also need permission by @cmf028 to relicense @gimhael's and @Kangz's shaders because of contributions.
> 
> Note: please do NOT merge this PR until every contributor to these files (i.e. legal owners) said they are OK with a relicensing.

Relicensing is fine with me as well.

illwieckz commented:
great :+1:

illwieckz commented:
all the commits in that PR are now mergeable, I'm just waiting for an answer from @DolceTriade about another file to know if if I can add this another file to the PR or not:

> @DolceTriade this file looks to be brand new file from you:
> 
> - [src/engine/null/null_renderer.cpp](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/null/null_renderer.cpp) (see [`f854e45`](https://github.com/DaemonEngine/Daemon/commit/f854e454ff5ee771c491b1e5e5c86bd0d518c400#diff-d2a454ec3022c55a7061696c2ffd8c32))
> 
> Do you confirm?

illwieckz commented:
I'm merging this, if we can relicense the other file there will still be time to make another commit
